tests/test_app.py

```diff
@@ -32,6 +32,16 @@ async def test_notebook_handler(notebooks, jp_fetch):
         html = r.body.decode()
         assert "Jupyter Notebook" in html
 
+    redirected_url = None
+
+    def redirect(self, url):
+        nonlocal redirected_url
+        redirected_url = url
+
+    NotebookHandler.redirect = redirect
+    await jp_fetch("notebooks", "jlab_test_notebooks")
+    assert redirected_url == "/a%40b/tree/jlab_test_notebooks"
```

Comment:
Thanks for adding this test!

In addition to this test, maybe it could be interesting to add a UI test to mimic real user interactions: https://github.com/jupyter/notebook/tree/main/ui-tests

But we can track that in a separate issue :+1: